app/main.py

```diff
@@ -1,214 +1,48 @@
-        "hp": 75,
-        "armour": [
-            {
-                "part": "helmet",
-                "protection": 15,
-            },
-            {
-                "part": "breastplate",
-                "protection": 20,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Two-handed Sword",
-            "power": 55,
-        },
-        "potion": None,
-    },
-    "mordred": {
-        "name": "Mordred",
-        "power": 30,
-        "hp": 90,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 15,
-            },
-            {
-                "part": "boots",
-                "protection": 10,
-            }
-        ],
-        "weapon": {
-            "name": "Poisoned Sword",
-            "power": 60,
-        },
-        "potion": {
-            "name": "Berserk",
-            "effect": {
-                "power": +15,
-                "hp": -5,
-                "protection": +10,
-            }
-        }
-    },
-    "red_knight": {
-        "name": "Red Knight",
-        "power": 40,
-        "hp": 70,
-        "armour": [
-            {
-                "part": "breastplate",
-                "protection": 25,
-            }
-        ],
-        "weapon": {
-            "name": "Sword",
-            "power": 45
-        },
-        "potion": {
-            "name": "Blessing",
-            "effect": {
-                "hp": +10,
-                "power": +5,
-            }
-        }
+from app.knights.knight import Knight
+from app.knights.equipment import Armor, Weapon, Potion
+from app.battle.fight import fight
+
+
+def create_knights(config: dict) -> dict:
+    knights = {}
+    for knight_name, attributes in config.items():
+        armor = [Armor(
+            a["part"], a["protection"]
+        ) for a in attributes["armour"]]
+        weapon = Weapon(
+            attributes["weapon"]["name"],
+            attributes["weapon"]["power"]
+        )
+        potion = Potion(
+            attributes["potion"]["name"],
+            attributes["potion"]["effect"]
+        ) if attributes["potion"] else None
+        knights[knight_name] = Knight(
+            attributes["name"],
+            attributes["power"],
+            attributes["hp"],
+            armor,
+            weapon,
+            potion
+        )
+    return knights
+
+
+def battle(knights_config):
+    knights = create_knights(knights_config)
+    # Execute Lancelot vs Mordred and Arthur vs Red Knight
+    result1 = fight(knights["lancelot"], knights["mordred"])
+    result2 = fight(knights["arthur"], knights["red_knight"])
+
+    # Combine results
+    return {**result1, **result2}
+
+
+if __name__ == "__main__":
+    KNIGHTS = {
+        "lancelot": {...},  # Insert knight configurations
+        "arthur": {...},
+        "mordred": {...},
+        "red_knight": {...}
```

Comment:
Ensure that the knight configurations are correctly filled in with the necessary attributes such as 'name', 'power', 'hp', 'armour', 'weapon', and 'potion'. Without these configurations, the `create_knights` function will not be able to instantiate the `Knight` objects properly.